src/Workspaces/Core/Portable/TodoComments/TodoCommentOptions.cs

```diff
@@ -2,27 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Immutable;
-using System.Composition;
-using Microsoft.CodeAnalysis.Host.Mef;
+using System.Runtime.Serialization;
 using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.Options.Providers;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.TodoComments
+namespace Microsoft.CodeAnalysis.TodoComments
 {
-    [ExportSolutionOptionProvider, Shared]
-    internal sealed class TodoCommentOptions : IOptionProvider
+    [DataContract]
+    internal readonly record struct TodoCommentOptions(
+        [property: DataMember(Order = 0)] string TokenList = "")
     {
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public TodoCommentOptions()
+            : this(TokenList: "")
         {
         }
 
-        ImmutableArray<IOption> IOptionProvider.Options { get; } = ImmutableArray.Create<IOption>(
-            TokenList);
-
-        public static readonly Option2<string> TokenList = new(nameof(TodoCommentOptions), nameof(TokenList), defaultValue: "");
+        public static readonly TodoCommentOptions Default = new();
```

Comment:
whihc constructor does this call? :'(   this part of the language confuses me.

Reply:
Yes, we did not end up in a good spot here with structs. Not sure if this can be fixed, probably not. This will call the explicitly defined parameterless ctor. The one with optional parameters is different one. That's actually why I'm defining the parameterless one so that one does not end up in a trap of calling `new T()` and ending up with `default(T)` instead of the default values.

Also, something like this would be useful:

```C#
readonly record struct R(optional T X = <expr>);
```

which would translate to:

```C#
readonly struct R
{
  public readonly T X;
  R(T? X = default) { this.X = X ?? <expr>; }
}
```